include/RAJA/policy/cuda/atomic.hpp

```diff
@@ -43,596 +51,549 @@ namespace RAJA
-template <>
-RAJA_INLINE __device__ unsigned cuda_atomicDec<unsigned>(unsigned volatile *acc)
-{
-  return ::atomicSub((unsigned *)acc, (unsigned)1);
-}
+/*!
+ * Atomic min/max
+ */
+using cuda_atomicMinMax_builtin_types = list<
+      int
+     ,unsigned int
+#if __CUDA_ARCH__ >= 500
+     ,long long int
+     ,unsigned long long int
 #endif
+    >;
 
 
-#if __CUDA_ARCH__ >= 200
-template <typename T>
-RAJA_INLINE __device__ T cuda_atomicAnd(T volatile *acc, T value)
+/*!
+ * Atomic min
+ */
+template <typename T,
+          enable_if_is_none_of<T, cuda_atomicMinMax_builtin_types>* = nullptr>
+RAJA_INLINE __device__ T cuda_atomicMin(T *acc, T value)
 {
-  return cuda_atomic_CAS_oper(acc, [=] __device__(T a) {
-    return a & value;
-  });
+  return cuda_atomicCAS(acc,
+                        [value] (T old) {
+                          return value < old ? value : old;
+                        },
+                        [value] (T current) {
+                          return current < value;
+                        });
 }
 
-// 32-bit signed atomicAnd support by CUDA
-template <>
-RAJA_INLINE __device__ int cuda_atomicAnd<int>(int volatile *acc,
-                                          int value)
+template <typename T,
+          enable_if_is_any_of<T, cuda_atomicMinMax_builtin_types>* = nullptr>
+RAJA_INLINE __device__ T cuda_atomicMin(T *acc, T value)
 {
-  return ::atomicAnd((int *)acc, value);
+  return ::atomicMin(acc, value);
 }
 
 
-// 32-bit unsigned atomicAnd support by CUDA
-template <>
-RAJA_INLINE __device__ unsigned cuda_atomicAnd<unsigned>(unsigned volatile *acc,
-                                                    unsigned value)
+/*!
+ * Atomic max
+ */
+template <typename T,
+          enable_if_is_none_of<T, cuda_atomicMinMax_builtin_types>* = nullptr>
+RAJA_INLINE __device__ T cuda_atomicMax(T *acc, T value)
 {
-  return ::atomicAnd((unsigned *)acc, value);
+  return cuda_atomicCAS(acc,
+                        [value] (T old) {
+                          return old < value ? value : old;
+                        },
+                        [value] (T current) {
+                          return value < current;
+                        });
 }
-#endif
 
-// 64-bit unsigned atomicAnd support by CUDA sm_35 and later
-#if __CUDA_ARCH__ >= 350
-template <>
-RAJA_INLINE __device__ unsigned long long cuda_atomicAnd<unsigned long long>(
-    unsigned long long volatile *acc,
-    unsigned long long value)
+template <typename T,
+          enable_if_is_any_of<T, cuda_atomicMinMax_builtin_types>* = nullptr>
+RAJA_INLINE __device__ T cuda_atomicMax(T *acc, T value)
 {
-  return ::atomicAnd((unsigned long long *)acc, value);
+  return ::atomicMax(acc, value);
 }
-#endif
 
-#if __CUDA_ARCH__ >= 200
-template <typename T>
-RAJA_INLINE __device__ T cuda_atomicOr(T volatile *acc, T value)
+
+/*!
+ * Atomic increment/decrement with reset
+ */
+using cuda_atomicIncDecReset_builtin_types = list<
```

Comment:
is this camp 'list'? Please qualify with namespace.

Reply:
Fixed.